### PR TITLE
feat: parser to parse test files and load testcases

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

File: literal/utils.go [hunk 1]
```diff
@@ -14,8 +17,8 @@ func NewBool(value bool) (expr.Literal, error) {
 	return expr.NewPrimitiveLiteral[bool](value, false), nil
 }
 
-func NewInt8(value int8) (expr.Literal, error) {
-	return expr.NewPrimitiveLiteral[int8](value, false), nil
+func NewInt8(value int8) expr.Literal {
+	return expr.NewPrimitiveLiteral[int8](value, false)
 }
 
 func NewInt16(value int16) (expr.Literal, error) {
```
Comment (EpsilonPrime) [inline literal/utils.go:24]: If the other signatures are changing, should these change too?

File: literal/utils_test.go [hunk 2]
```diff
@@ -738,3 +825,138 @@ func TestNewPrecisionTimestampTz(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		value   string
+		want    expr.Literal
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{"empty", "", nil, assert.Error},
+		{"1970-01-01", "1970-01-01", expr.NewPrimitiveLiteral(types.Date(0), false), assert.NoError},
+		{"2021-01-01", "2021-01-01", expr.NewPrimitiveLiteral(types.Date(18628), false), assert.NoError},
+		{"2021-01-01T00:00:00Z", "2021-01-01T00:00:00Z", nil, assert.Error},
```
Comment (EpsilonPrime) [inline literal/utils_test.go:924]: We recently added support for strings like this elsewhere -- should we also support them here?

File: testcases/parser/baseparser/functestcaseparser_listener.go
Comment (EpsilonPrime) [inline testcases/parser/baseparser/functestcaseparser_listener.go:1]: Do we need both the listener and visitor version?

File: literal/utils.go [hunk 3]
```diff
@@ -14,8 +17,8 @@ func NewBool(value bool) (expr.Literal, error) {
 	return expr.NewPrimitiveLiteral[bool](value, false), nil
 }
 
-func NewInt8(value int8) (expr.Literal, error) {
-	return expr.NewPrimitiveLiteral[int8](value, false), nil
+func NewInt8(value int8) expr.Literal {
```
Comment (EpsilonPrime) [inline literal/utils.go:20]: Nit:  It's way too late but in the future consider using FromInt8 instead as this is a conversion and not the construction of a pristine value.

**Review by EpsilonPrime — COMMENTED** [review]
There are a large number of positive paths in visitor.go without test coverage.

File: literal/utils.go (hunk 3 above)
Comment (scgkiran) [reply literal/utils.go:20]: ok, noted!

File: testcases/parser/baseparser/functestcaseparser_listener.go
Comment (scgkiran) [reply testcases/parser/baseparser/functestcaseparser_listener.go:1]: removed listener

File: literal/utils_test.go (hunk 2 above)
Comment (scgkiran) [reply literal/utils_test.go:909]: can you please give reference?

File: literal/utils.go (hunk 1 above)
Comment (scgkiran) [reply literal/utils.go]: Yes, Fixed them.

**Comment by scgkiran** [comment]
> There are a large number of positive paths in visitor.go without test coverage.

Fixed most of them. Rest of them need a grammar change, I missed few literal types for test cases.

File: literal/utils_test.go (hunk 2 above)
Comment (EpsilonPrime) [reply literal/utils_test.go:909]: I missed this section is testing dates and not timestamps.  The test is fine as is.

**Review by EpsilonPrime — APPROVED** [review]

**Pull request merged**